### PR TITLE
[REF] l10n_it_edi: move_type hook

### PR DESCRIPTION
**Pull request by jco-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 1]
```diff
@@ -173,6 +173,16 @@ def _l10n_it_get_document_type(self, invoice):
     def _l10n_it_is_simplified_document_type(self, document_type):
         return document_type in ['TD07', 'TD08', 'TD09']
 
+    def _l10n_it_get_move_type(self, document_type):
+        move_type = 'in_invoice'
+        document_type = elements[0].text if elements else ''
```
Comment (lordkrandel) [inline addons/l10n_it_edi/models/account_edi_format.py:178]: 178 to be removed

**Review by lordkrandel — CHANGES_REQUESTED** [review]
I do agree with the function extraction.
Will need update for B2G for all the other document types that we'll have to handle.

```
TD01 invoice
TD02 downpayment on invoice
TD03 downpayment on fee
TD04 credit note
TD05 debit note
TD06 fee
TD16 invoice integration (or self-invoice) for internal reverse charge
TD17 invoice integration (or self-invoice) for foreign purchase
TD18 invoice integration for intra-UE goods
TD19 invoice integration (or self-invoice) for purchase of goods ex art.17 c.2 DPR 633/72
TD20 self-invoice for the settlement and integration of invoices (ex art.6 c.8 d.lgs. 471/97 o art.46 c.5 D.L. 331/93)
TD21 self-invoice for plafond overrun
TD22 Removal of goods from VAT storage
TD23 Removal of goods from VAT storage with VAT payment
TD24 Deferred invoice - art. 21, comma 4, lett. a
TD25 Deferred invoice - art. 21, comma 4, terzo periodo lett. b
TD26 Sale of assets and internal transfers (ex art.36 DPR 633/72)
TD27 Invoice for self-consumption and free transfer without recourse
```

**Comment by OpenCode** [comment]
@jco-odoo Just for information, this PR is born from this one: https://github.com/odoo/odoo/pull/91327
We can use the original one to discuss about the implementation.

**Review by Levizar — CHANGES_REQUESTED** [review]

**Comment by OpenCode** [comment]
> be aware

Thank you

**Comment by jco-odoo** [comment]
@OpenCode I think it is outdated now.  Let us know if it would not still solve your problem

**Pull request closed without merging**